src/lib/core/tests/TestCHIPTLV.cpp

```diff
@@ -237,13 +240,13 @@ void TestDupString(nlTestSuite * inSuite, TLVReader & reader, Tag tag, const cha
     size_t expectedLen = strlen(expectedVal);
     NL_TEST_ASSERT(inSuite, reader.GetLength() == expectedLen);
 
-    chip::Platform::ScopedMemoryBuffer<char> valBuffer;
-    char * val = valBuffer.Alloc(expectedLen + 1).Get();
-
+    char * val     = nullptr;
     CHIP_ERROR err = reader.DupString(val);
     NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
-
+    NL_TEST_ASSERT(inSuite, val != nullptr);
```

Comment:
@mrjerryjohns  - this was an API misunderstanding it seems, both for DupString and DupBytes.

I do not see usages of DupString/Bytes beyond tests. Should we instead remove these methods?